gen/testdata/collision/types.go

```diff
@@ -32,6 +32,12 @@ func (v *LittlePotatoe) FromWire(w wire.Value) error {
 	return err
 }
 
+func (lhs LittlePotatoe) Equals(rhs LittlePotatoe) bool {
+	l := (int64)(lhs)
+	r := (int64)(rhs)
+	return (l == r)
```

Comment:
I don't think we need this cast here, you can just do `return (l == r)`. Any reason why we do the conversion to an `int64`?